scipy/_lib/_util.py

```diff
@@ -11,6 +11,53 @@
 import numpy as np
 
 
+def _valarray(shape, value=np.nan, typecode=None):
+    """Return an array of all value.
+    """
+
+    out = np.ones(shape, dtype=bool) * value
+    if typecode is not None:
+        out = out.astype(typecode)
+    if not isinstance(out, np.ndarray):
+        out = np.asarray(out)
+    return out
```

Comment:
I'm wondering why it's written this way rather than just `np.asarray(np.empty(shape).fill(value), dtype=dtype)` (I know this is pre-existing code, so this comment is not meant to slow down this PR).